### PR TITLE
Threshold stats unit tests

### PR DESCRIPTION
**Pull request by pritchyspritch**
This PR adds unit tests for get_metric_statistics which is the main method used to calculate the threshold. The get_metric_threshold was difficult to unit test because of its mixed use of different boto clients.

### CODE REVIEW

File: .secrets.baseline [hunk 1]
```diff
@@ -45,7 +45,38 @@
...
+      },
+      {
+        "hashed_secret": "01be741099a377ecf0810426de0ec62ada610a4c",
+        "is_secret": false,
+        "is_verified": false,
+        "line_number": 307,
+        "type": "Base64 High Entropy String"
+      }
+    ]
+  },
```
Comment (0atman) [inline .secrets.baseline]: :ok_hand:

**Comment by pritchyspritch** [comment]
> Tests fail for me, using `make test` in a fresh venv

That's weird, I just added a fresh virtualenv on mine and it passed...

**Comment by pritchyspritch** [comment]
> > Tests fail for me, using `make test` in a fresh venv
> 
> That's weird, I just added a fresh virtualenv on mine and it passed...and the code it's complaining about isn't part of this PR...

File: .secrets.baseline (hunk 1 above)
Comment (pritchyspritch) [reply .secrets.baseline]: > Tests fail for me, using `make test` in a fresh venv

Try now. (I closed the PR completely by accident, pressed the wrong button) 🙄

**Pull request merged**